app/viewmodels/models/MemberState.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.models
+
+import play.api.libs.json.{Format, JsError, JsResult, JsString, JsSuccess, JsValue}
+import utils.WithName
+
+sealed trait MemberState {
+  val name: String
+}
+
+object MemberState {
+  case object Active extends WithName("Active") with MemberState
+
+  case object Deleted extends WithName("Deleted") with MemberState
```

Comment:
This is a pre-requisite for soft delete, but it hasn't been used yet. Is this understanding correct?

Reply:
Yup thats right - I started to add it here before splitting the tickets